src/uk/gov/defra/ffc/Provision.groovy

```diff
@@ -25,6 +27,95 @@ class Provision implements Serializable {
+           // removing the schema removes the database migrations within that schema, so is unneccessary
+           ctx.sh("docker-compose -p $repoName-$pr -f docker-compose.migrate.yaml run --no-deps schema-down")
+        }
+      }
+    }
+  }
+
+  private static def getMigrationFiles(ctx, destinationFolder){
+    def resourcePath = 'uk/gov/defra/ffc/migration'
+    ctx.sh(" rm -rf $destinationFolder")
+    getResourceScript(ctx, "$resourcePath/scripts", 'schema-up', "$destinationFolder/scripts")
+    getResourceScript(ctx, "$resourcePath/scripts", 'schema-down', "$destinationFolder/scripts")
+    getResourceFile(ctx, resourcePath, 'docker-compose.migrate.yaml', destinationFolder)
+    getResourceFile(ctx, "$resourcePath/changelog", 'schema.changelog.xml', "$destinationFolder/changelog")
+  }
+
+  private static def getResourceFile(ctx, resourcePath, filename, destinationFolder, makeExecutable = false){
+    def fileContent = ctx.libraryResource("$resourcePath/$filename")
+    ctx.writeFile(file: "$destinationFolder/$filename", text: fileContent, encoding: "UTF-8")
+    if (makeExecutable) {
+      ctx.sh("chmod 777 ./$destinationFolder/$filename")
+    }
+    ctx.echo "written $filename to $destinationFolder"
+  }
+
+  private static def getResourceScript(ctx, resourcePath, filename, destinationFolder){
+    getResourceFile(ctx, resourcePath, filename, destinationFolder, true)
+  }
+
+  private static def getMigrationEnvVars(ctx, environment, repoName, pr) {
+    def searchKeys = [
+      'POSTGRES_HOST',
+      'POSTGRES_ADMIN_USERNAME',
+      'POSTGRES_ADMIN_PASSWORD',
+      'POSTGRES_SCHEMA_PASSWORD'
+    ]
+    def appConfigPrefix = environment + '/'
+    def appConfigValues = Utils.getConfigValues(ctx, searchKeys, appConfigPrefix, Utils.defaultNullLabel, false)
+    appConfigValues['POSTGRES_ADMIN_PASSWORD'] = escapeQuotes(appConfigValues['POSTGRES_ADMIN_PASSWORD'])
+    appConfigValues['POSTGRES_SCHEMA_PASSWORD'] = escapeQuotes(appConfigValues['POSTGRES_SCHEMA_PASSWORD'])
+
+    def migrationEnvVars = appConfigValues.collect { "$it.key=$it.value" }
+
+    def schemaName = repoName.replace('-','_') + pr
+    def schemaRole = repoName.replace('-','_') + pr + "role"
+    def schemaUser = getSchemaUserWithHostname(schemaRole, appConfigValues['POSTGRES_HOST'])
+    def databaseName = repoName.replace('-','_').replace('_service', '')
+
+    migrationEnvVars.add("POSTGRES_SCHEMA_ROLE=$schemaRole")
+    migrationEnvVars.add("POSTGRES_SCHEMA_USERNAME=$schemaUser")
+    migrationEnvVars.add("POSTGRES_SCHEMA_NAME=$schemaName")
+    migrationEnvVars.add("POSTGRES_DB=$databaseName")
+
+    return migrationEnvVars
+  }
+
+  private static def escapeQuotes(value) {
+    return value.replace("\"", "\\\"")
+  }
+
+  private static getSchemaUserWithHostname(schemaRole, dbServer) {
+    // string includes quotes, so remove them and escape '.' as split takes a regex
+    def dbServerSplit = dbServer.replace("\"","").split('\\.')
```

Comment:
The code for extracting the name of the server from `POSTGRES_HOST` is fairly minimal, however, I was thinking it could be removed altogether if a key was added to app config for the name e.g. `POSTGRES_SERVER_NAME`. I'm not sure which is better. If the name were to change and it might mean a change to this code whereas if there was a app config setting for it, just that could be updated. What do you think?

Reply:
I did consider this, but came to the conclusion it would be more error prone to have duplication in the app config keys and have to maintain them, rather than split out the hostname from the full server address in the code.